File: src/tests/test_quickSort.py
from typing import List

class QuickSort:
    def sort(self, arr: int, low: int, high: int) -> None:
        if low < high:
            pi = None
            pi = self.partition(arr, low, high)
            self.sort(arr, low, pi - 1)
            self.sort(arr, pi + 1, high)
    def partition(self, arr: int, low: int, high: int) -> int:
        pivot = None
        pivot = arr[high]
        i = None
        i = low - 1
        for j in range(low, high):
            if arr[j] < pivot:
                (i:=i+1)
                temp = None
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
        temp = None
        temp = arr[i + 1]
        arr[i + 1] = arr[high]
        arr[high] = temp
        return i + 1
    def __init__(self):
            pass
def main():
    arr = [0] * 100
    size = None
    size = 8
    print("Input the array length: ", sep='', end='')
    size = int(input())
    print("Input integers: ", sep='', end='')
    for i in range(0, size):
        arr[i] = int(input())
    print("Original array: ", sep='', end='')
    for i in range(0, size):
        print(arr[i], " ", sep='', end='')
    print("\n", sep='', end='')
    sorter = QuickSort()
    sorter.sort(arr, 0, size - 1)
    print("Sorted array: ", sep='', end='')
    for i in range(0, size):
        print(arr[i], " ", sep='', end='')
    print("\n", sep='', end='')
    return 0

if __name__ == '__main__':
    main()

File: src/tests/test_quickSort.cpp
#include <iostream>

class QuickSort {
public:
    void sort(int* arr, int low, int high) {
        if (low < high) {
            int pi = partition(arr, low, high);

            sort(arr, low, pi - 1);
            sort(arr, pi + 1, high);
        }
    }

private:
    int partition(int* arr, int low, int high) {
        int pivot = arr[high]; 
        int i = low - 1;      

        for (int j = low; j < high; ++j) {
            if (arr[j] < pivot) {
                ++i;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }

        int temp = arr[i + 1];
        arr[i + 1] = arr[high];
        arr[high] = temp;

        return i + 1;
    }
};

int main() {
    int arr[100];
    int size = 8;

    // 输入数组长度
    std::cout << "Input the array length: ";
    std::cin >> size;

    // 输入整数
    std::cout << "Input integers: ";
    for (int i = 0; i < size; i++) {
        std::cin >> arr[i];
    }

    std::cout << "Original array: ";
    for (int i = 0; i < size; ++i) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    QuickSort sorter;
    sorter.sort(arr, 0, size - 1);

    std::cout << "Sorted array: ";
    for (int i = 0; i < size; ++i) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}
